src/CMakeLists.txt

```diff
@@ -303,29 +360,42 @@ if(CMAKE_CROSSCOMPILING)
     # Select the smaller version of libc called nano.
     target_compile_options(${elf} PRIVATE --specs=nano.specs)
     target_link_libraries(${elf} PRIVATE --specs=nano.specs)
-    if(NOT firmware STREQUAL "firmware-semihosting")
+    string(FIND ${firmware} semihosting SEMIHOSTING_FOUND)
+    if(SEMIHOSTING_FOUND EQUAL -1)
       target_compile_options(${elf} PRIVATE --specs=nosys.specs)
       target_link_libraries(${elf} PRIVATE --specs=nosys.specs)
+    else()
+      # Select an implementation of the system calls that can communicate with the debugger
+      target_compile_options(${elf} PRIVATE --specs=rdimon.specs)
+      target_link_libraries(${elf} PRIVATE --specs=rdimon.specs)
     endif()
   endforeach(firmware)
 
   target_sources(firmware.elf PRIVATE firmware.c)
-  target_compile_definitions(firmware.elf PRIVATE APP_BTC APP_LTC APP_ETH APP_U2F)
-  target_sources(firmware.elf PRIVATE ${FIRMWARE-U2F-SOURCES})
+  target_compile_definitions(firmware.elf PRIVATE PRODUCT_BITBOX_MULTI APP_BTC APP_LTC APP_ETH APP_U2F)
+  target_sources(firmware.elf PRIVATE ${FIRMWARE-U2F-SOURCES} ${PLATFORM-BITBOX02-SOURCES})
 
   target_sources(firmware-semihosting.elf PRIVATE firmware.c)
-  target_sources(firmware-semihosting.elf PRIVATE ${FIRMWARE-U2F-SOURCES})
+  target_sources(firmware-semihosting.elf PRIVATE ${FIRMWARE-U2F-SOURCES} ${PLATFORM-BITBOX02-SOURCES})
   # Select an implementation of the system calls that can communicate with the debugger
-  target_compile_options(firmware-semihosting.elf PRIVATE --specs=rdimon.specs)
-  target_link_libraries(firmware-semihosting.elf PRIVATE --specs=rdimon.specs)
-  target_compile_definitions(firmware-semihosting.elf PRIVATE APP_BTC APP_LTC APP_ETH APP_U2F)
+  target_compile_definitions(firmware-semihosting.elf PRIVATE PRODUCT_BITBOX_MULTI APP_BTC APP_LTC APP_ETH APP_U2F)
   target_compile_definitions(firmware-semihosting.elf PRIVATE SEMIHOSTING)
 
   target_sources(firmware-btc.elf PRIVATE firmware.c)
-  target_compile_definitions(firmware-btc.elf PRIVATE FIRMWARE_BTC_ONLY APP_BTC)
+  target_compile_definitions(firmware-btc.elf PRIVATE PRODUCT_BITBOX_BTCONLY APP_BTC)
+  target_sources(firmware-btc.elf PRIVATE ${PLATFORM-BITBOX02-SOURCES})
+
+  target_sources(firmware-base.elf PRIVATE bitboxbase/bitboxbase.c)
```

Comment:
`firmware-bitboxbase` please. sorry i keep bothering you about it :P

Reply:
ooh, thats quite a big change. I don't agree with it, but I can change it anyway.

Reply:
Why not agree? It's the same reasoning as the other places where you agreed (protobuf msgs, etc.). `firmware-base` sounds like some sort of foundation for other firmwares. I am a big advocate of explicit/descriptive naming.